back/reader.js

```diff
@@ -0,0 +1,83 @@
+const fs = require("fs");
+const util = require("util");
+
+// Convert fs methods to return promises
+const fsOpen = util.promisify(fs.open);
+const fsRead = util.promisify(fs.read);
+const fsClose = util.promisify(fs.close);
+const fsStat = util.promisify(fs.stat);
+
+const readFile = (path, start, onRead) => {
+    const readStream = fs.createReadStream(path, {start: start});
+
+    readStream.on("data", (chunk) => {
+        onRead(chunk.toString(), start + chunk.byteLength);
+    });
+
+    readStream.on("error", (err) => {
+        console.error("An error occurred:", err);
+    });
+};
+
+const readFromEnd = async (path, onRead, lines) => {
+    try {
+        const stats = await fsStat(path);
+        let position = stats.size;
+        const buffer = Buffer.alloc(1);
+        let lineCount = 0;
+        let content = "";
+        let fileDescriptor = await fsOpen(path, "r");
+
+        while (position > 0 && lineCount <= lines) {
+            position -= 1;
+            const {bytesRead} = await fsRead(
```

Comment:
can we optimize this further than reading only 1 byte at a time ?

Reply:
started reading 100 bytes at a time